server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTreeBelowSGCachedImpl.java

```diff
@@ -578,39 +592,78 @@ public boolean isEmptyInternalMNode(IMNode node) throws MetadataException {
     }
   }
 
+  @Override
+  public List<PartialPath> constructSchemaBlackList(PartialPath pathPattern)
+      throws MetadataException {
+    List<PartialPath> result = new ArrayList<>();
+    try (MeasurementUpdater updater =
+        new MeasurementUpdater(rootNode, pathPattern, store, false) {
+          @Override
+          protected void updateMeasurement(IMeasurementMNode node) throws MetadataException {
+            node.setPreDeleted(true);
+            updateMNode(node);
```

Comment:
Since the updated node won't be cache evicted until it been flushed to disk and currently the flush won't happen during traversing which takes the read lock, we need to consider the concurrency of flush and traverse for better memory control in future work.

Reply:
Sure. I added it to the Updater's TODO in case I forgot.